src/wavealign/data_collection/audio_property_sets_reader.py

```diff
@@ -0,0 +1,50 @@
+import os
+import traceback
+
+from wavealign.data_collection.audio_property_set_generator import (
+    AudioPropertySetGenerator,
+)
+from wavealign.data_collection.audio_property_set import AudioPropertySet
+from wavealign.data_collection.audio_file_finder import AudioFileFinder
+from wavealign.loudness_processing.window_size import WindowSize
+
+# TODO: Add progress bar
+
+
+class AudioPropertySetsReader:
+    def __init__(
+        self,
+        input_path: str,
+        window_size: WindowSize,
+        cache_manager=None,
```

Comment:
Use optional type annotation 
```suggestion
        cache_manager: CacheManage | None = None,
```

Reply:
Those optional annotations are not available in python 3.9. I'll create a TODO to fully move to 3.11.  Sorry for all the TODOs, we just arrived in Georgia and time is a little short. Maybe I can tackle some of those in the next days!